packages/build/src/tracing/main.ts

```diff
@@ -1,6 +1,5 @@
+import { HoneycombSDK } from '@honeycombio/opentelemetry-node'
 import { context, trace, propagation, SpanStatusCode, diag, DiagLogLevel, DiagLogger } from '@opentelemetry/api'
-import { OTLPTraceExporter } from '@opentelemetry/exporter-trace-otlp-grpc'
-import { HttpInstrumentation } from '@opentelemetry/instrumentation-http'
```

Comment:
Couldn't manage to setup auto instrumentation. Seems like it's now working for `esm` but it's fairly recent and requires us to pass additional `node` flags:
- https://github.com/open-telemetry/opentelemetry-js/pull/3698
- https://github.com/open-telemetry/opentelemetry-js/blob/dccd90603bf52ba3b52eced9e8a6489bbdc78388/examples/esm-http-ts/package.json#L10

I've been thinking a bit about this and I think in the future we might want to have this tracing initialisation happen in a separate module we can use via `node --require @netlify/tracing-initialisation ./call-neltify/build` (or `node --import` once we move to node 20 I believe?) since it would:
- Allow us to out of the box use the same initialisation process for any node process we startup in Buildbot
- Make `@netlify/build` and other packages only depend on `@opentelemetry/api` (see https://github.com/netlify/cli/pull/5826#issuecomment-1612587919)